compiler/modules/front.syntax/parseTypeDesc.bal

```diff
@@ -68,10 +68,20 @@ function parsePostfixTypeDesc(Tokenizer tok) returns TypeDesc|err:Syntax {
             td = bin;
         }
         else if tok.current() == "[" {
-            check tok.advance();
-            Position endPos = check tok.expectEnd("]");
-            ListTypeDesc list = { startPos, endPos, members: [], rest: td };
-            td = list;
+            SimpleConstExpr?[] dimensions = [];
+            Position? endPos = (); // Init with nil as we can't prove to the compiler that first iteration will always run.
```

Comment:
Can we use `while true` and check for tok.current() at the end, so we can avoid the cast?

Reply:
Because we have a `if` block before the initialization of `endPos` jBallerina is going to report an un-initialized variable.